Basics 1 (#31)
mlr-org/mlr3book: bookdown/index.Rmd

library("R6")
```

# Quickstart

As a 30-second introductory example, we will train a decision tree model on the first 120 rows of iris data set and make predictions on the final 30, measuring the accuracy of the trained model.

```{r}
library(mlr3)
task = mlr_tasks$get("iris")
learner = mlr_learners$get("classif.rpart")

# train a model of this learner for a subset of the task
learner$train(task, row_ids = 1:120)
# this is what the decision tree looks like
learner$model

predictions = learner$predict(task, row_ids = 121:150)
predictions
# accuracy of our model on the test set of the final 30 rows
predictions$score("classif.acc")
```

The code examples in this book use a few additional packages that are not installed by default if you install `r mlr_pkg("mlr3")`.

> To run the code here, install the `r gh_pkg("mlr-org/mlr3book")` package using the `r cran_pkg("remotes")` package:

Reviewer: ```suggestion
To run all examples in the book, install the `r gh_pkg("mlr-org/mlr3book")` package using the `r cran_pkg("remotes")` package:
```

```{r index-002, eval = FALSE}
remotes::install_github("mlr-org/mlr3book", dependencies = TRUE)
```

> You don't need to load `r mlr_pkg("mlr3book")` though, just load the `r mlr_pkg("mlr3")` package and you are good to go:

Reviewer: In fact all packages will be loaded in their respective chunks.
So this statement if only half true as is suggests that the user only needs to load _mlr3_.
I would write something like "all required packages will be loaded when required throughout the book".

Author: I don't understand your point -- the sentence is saying that to run the examples in the book, only the mlr3 package needs to be loaded and not the mlr3book package. Is that not true? Does the user need to load any other packages?

That more packages may be loaded later on is a technical detail that the user doesn't need to know here.

Reviewer 2: We should make this clear:

* Install mlr3book if you (a) want to compile the book yourself or (b) do not want to be bugged while reading that some required packages are not installed.
* Install mlr3 if you just want to give it a shot and don't mind that you occasionally have to `install.packages()`.

Either way, load `mlr3`.

Author: So do you agree with the current wording?

Reviewer 2: Fine for me.

Reviewer: Install _mlr3verse_ (?) if you just want to give it a shot and don't mind that you occasionally have to install.packages()

-> usually people need more than just _mlr3_ to get their work done

Reviewer: @larskotthoff 	what's your view on this?
I am not pushing on a change if you like the current wording, this is just a suggestion to make it more clear imo and also include the _mlr3verse_ package at this point.

Author: Is there any advantage to installing mlr3verse instead of mlr3book? If we want to go this way, why have a mlr3book package in the first place?

Reviewer: You only need to install half of the packages compared to mlr3book. See https://github.com/mlr-org/mlr3book/pull/30#issuecomment-514647772.

I do not see the point why we should force users to install the whole book dependencies.
Which user wants to run all the examples in the book? Users will always focus on specific sections.

And I would dislike seeing the _mlr3book_ being promoted as the "umbrella" package for the mlr3 pkg framework - this is not really its purpose.
We want to stay slim and lightweight and then suggest to install the mlr3book with all of its deps?
This does not make sense to me.
And yes, even if we only write such a "short" and "small" sentence in the README of the intro, the danger is high that this suggestion gets established and people install the whole dep chain.

The book package exists because 
1. Michel needed to write some wrappers for linking and building the book to work around some bookdown shortcomings
2. To have a quick way of installing all deps of the book (which are a lot)

Author: Ok, but then how do we handle this here? We should mention that to be able to run everything you should install the mlr3book package.

Reviewer: See my two comments/suggestions

> You don't need to load `r mlr_pkg("mlr3book")` though, just load the `r mlr_pkg("mlr3")` package and you are good to go:

Reviewer: ```suggestion
Most often you do not want to run all code blocks of the book but just some parts of it.
In these cases it is easiest to install the _mlr3verse_ pkg (a wrapper package for the core packages of the _mlr3_ package framework) and post-install the required packages of the respective sections you want to run.

```r
remotes::install_github("mlr-org/mlr3verse")
```

```
